node-metrics/src/service/data_state/node_identity.rs

```diff
@@ -0,0 +1,248 @@
+use super::LocationDetails;
+use espresso_types::FeeAccount;
+use hotshot_types::signature_key::BLSPubKey;
+use serde::{Deserialize, Serialize};
+use surf_disco::Url;
+
+/// [NodeIdentity] represents the identity of the node that is participating
+/// in the network.
+#[derive(Clone, PartialEq, Debug, Serialize, Deserialize)]
+pub struct NodeIdentity {
+    pub(crate) public_key: BLSPubKey,
+    pub(crate) name: Option<String>,
+    pub(crate) wallet_address: Option<FeeAccount>,
+    pub(crate) public_url: Option<Url>,
+    pub(crate) company: Option<String>,
+    pub(crate) company_website: Option<Url>,
+    pub(crate) location: Option<LocationDetails>,
+    pub(crate) operating_system: Option<String>,
+    pub(crate) node_type: Option<String>,
+    pub(crate) network_type: Option<String>,
+}
+
+impl NodeIdentity {
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        public_key: BLSPubKey,
+        name: Option<String>,
+        wallet_address: Option<FeeAccount>,
+        public_url: Option<Url>,
+        company: Option<String>,
+        company_website: Option<Url>,
+        location: Option<LocationDetails>,
+        operating_system: Option<String>,
```

Comment:
Oh interesting.  Is there any way to get the version number with it as well?  Do we even want the version numbers with the operating system name, or just the operating system?

Reply:
Hmm that's a fair point. Version information would be nice to have. I found [this crate](https://docs.rs/sys-info/latest/sys_info/fn.os_release.html) which seems to give you that. But I'm also not opposed to just specifying it manually since this crate looks a little messy

Reply:
The `std::consts::env::OS` has been populated on the `sequencer` side.  The `NodeIdentity` only gets populated from scraping the `metrics` from the `status` page.  As this should refer to the `operating_system` of other `Node`s, I don't think this needs to be changed here, but it has been utilized in the appropriate play in the `sequencer` directly:

Reference e2a2180bd9e55278ee68b610aee33a19ae20f951.